include/node.tcc

```diff
@@ -599,3 +599,22 @@ void mpm::Node<Tdim, Tdof,
                                       separation_vector, Tdim);
   }
 }
+
+//! Compute multimaterial normal unit vector
+template <unsigned Tdim, unsigned Tdof, unsigned Tnphases>
+void mpm::Node<Tdim, Tdof,
+               Tnphases>::compute_multimaterial_normal_unit_vector() {
+  // Iterate over all materials in the material_ids set
+  std::lock_guard<std::mutex> guard(node_mutex_);
+  for (auto mitr = material_ids_.begin(); mitr != material_ids_.end(); ++mitr) {
+    // calculte the normal unit vector
+    const auto& domain_gradient =
+        property_handle_->property("domain_gradients", prop_id_, *mitr, Tdim);
+    const auto& normal_unit_vector =
+        (1 / domain_gradient.norm()) * domain_gradient;
```

Comment:
I would check if `domain_gradient.norm()` if not zero, and no need to add `&` in `auto`.
```suggestion
VectorDim normal;
if (domain_gradient.norm() > std::numeric_limits<double>::epsilon())
   normal = (1 / domain_gradient.norm()) * domain_gradient;
else
   normal.setZero();
```

Reply:
Can we use `normal = domain_gradient.normalize() instead?

Reply:
I did not add the else condition in my next commit because I initialised the normal vector with VectorDim::Zero() as @ezrayst suggested. Thank you for the other change requests!